core/src/Cache/TokenCacheSerializeHelper.cs

```diff
@@ -59,6 +59,8 @@ internal static void DeserializeUnifiedCache(TokenCacheAccessor tokenCacheAccess
                 return;
             }
 
+            tokenCacheAccessor.Clear();
```

Comment:
Shouldn't this go to before if (cacheDict is empty) ?

Reply:
oh, right - will fix it

Reply:
Good Catch